precompiles/vesting/integration_test.go

```diff
@@ -409,367 +429,456 @@ var _ = Describe("Interacting with the vesting extension", Ordered, func() {
+				funderInitialBal, err := s.grpcHandler.GetBalance(funderKey.AccAddr, s.bondDenom)
+				Expect(err).To(BeNil())
 
 				// Build and execute the tx to fund the vesting account from a smart contract
-				fundVestingAccArgs := s.BuildCallArgs(callType, contractAddr).
-					WithMethodName(vesting.FundVestingAccountMethod).
-					WithArgs(
-						funderContractAddr,
-						toAddr,
-						uint64(time.Now().Unix()),
-						defaultPeriods,
-						emptyPeriods,
-					)
-				_, _, err = contracts.CallContractAndCheckLogs(s.ctx, s.app, fundVestingAccArgs, execRevertedCheck)
-				Expect(err).To(HaveOccurred())
+				// Build and execute the tx to fund the vesting account from a smart contract.
+				callArgs, txArgs := s.BuildCallArgs(callType, vestingCallerAddr)
+				txArgs.GasPrice = gasPrice.BigInt()
+				callArgs.MethodName = vesting.FundVestingAccountMethod
+				callArgs.Args = []interface{}{
+					contractCounterAddr,
+					vestingAccKey.Addr,
+					uint64(time.Now().Unix()),
+					defaultPeriods,
+					emptyPeriods,
+				}
 
-				// check that tx signer's balance is reduced by the fees paid
-				txSenderFinalBal := s.app.BankKeeper.GetBalance(s.ctx, s.address.Bytes(), s.bondDenom)
-				Expect(txSenderFinalBal.Amount.LTE(txSenderInitialBal.Amount)).To(BeTrue())
+				res, _, err := s.factory.CallContractAndCheckLogs(funderKey.Priv, txArgs, callArgs, execRevertedCheck)
+				Expect(err).NotTo(HaveOccurred())
 
-				// the balance of the contract that calls the precompile should remain 0
-				contractFinalBal := s.app.BankKeeper.GetBalance(s.ctx, contractAddr.Bytes(), s.bondDenom)
-				Expect(contractFinalBal.Amount).To(Equal(sdk.ZeroInt()))
+				fees := gasPrice.MulRaw(res.GasUsed)
 
-				// the balance of the funder contract should remain unchanged
-				funderContractFinalBal := s.app.BankKeeper.GetBalance(s.ctx, funderContractAddr.Bytes(), s.bondDenom)
-				Expect(funderContractFinalBal.Amount).To(Equal(funderContractInitialAmt))
+				funderFinalBal, err := s.grpcHandler.GetBalance(funderKey.AccAddr, s.bondDenom)
+				Expect(err).To(BeNil())
+
+				Expect(funderFinalBal.Balance.Amount).To(Equal(funderInitialBal.Balance.Amount.Sub(fees)), "expected funder to have initial balance minus fees")
+				// // check that tx signer's balance is reduced by the fees paid
+				// txSenderFinalBal := s.app.BankKeeper.GetBalance(s.ctx, s.address.Bytes(), s.bondDenom)
+				// Expect(txSenderFinalBal.Amount.LTE(txSenderInitialBal.Amount)).To(BeTrue())
+				//
+				// // the balance of the contract that calls the precompile should remain 0
+				// contractFinalBal := s.app.BankKeeper.GetBalance(s.ctx, contractAddr.Bytes(), s.bondDenom)
+				// Expect(contractFinalBal.Amount).To(Equal(sdk.ZeroInt()))
+				//
+				// // the balance of the funder contract should remain unchanged
+				// funderContractFinalBal := s.app.BankKeeper.GetBalance(s.ctx, funderContractAddr.Bytes(), s.bondDenom)
+				// Expect(funderContractFinalBal.Amount).To(Equal(funderContractInitialAmt))
```

Comment:
Should we uncomment this or remove it?

Reply:
I removed the entire test because it was using a flow unavailable with the new test refactoring.